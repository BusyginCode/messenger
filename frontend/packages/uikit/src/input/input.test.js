import React from 'react'
import { mount } from 'enzyme'
import Input from './input'

describe('uikit/Input', () => {
  it('should renders correctly', () => {
    const onChange = jest.fn()
    const onFocus = jest.fn()
    const input = mount(<Input name="someInput" onChange={onChange} onFocus={onFocus} value={'some text'} />)

    expect(input).toMatchSnapshot()

    input
      .find('input')
      .simulate('change')
      .simulate('focus')

    expect(onChange).toHaveBeenCalledWith({
      target: {
        name: 'someInput',
        value: 'some text'
      }
    })

    expect(onFocus).toHaveBeenCalled()
  })

  it('should renders correctly with prependValue', () => {
    const onChange = jest.fn()
    const onFocus = jest.fn()
    const input = mount(
      <Input name="someInput" onChange={onChange} onFocus={onFocus} prependValue={'steam://'} value={'some text'} />
    )

    expect(input).toMatchSnapshot()

    input
      .find('input')
      .simulate('change')
      .simulate('focus')

    expect(onChange).toHaveBeenCalledWith({
      target: {
        name: 'someInput',
        value: 'some text'
      }
    })

    expect(onFocus).toHaveBeenCalled()
  })
})
