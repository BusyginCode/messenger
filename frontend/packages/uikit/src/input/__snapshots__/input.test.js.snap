// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Input should renders correctly 1`] = `
<Input
  intent="default"
  name="someInput"
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  size="l"
  type="text"
  value="some text"
>
  <div
    className="wrapper size-l"
  >
    <input
      className="input default"
      name="someInput"
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value="some text"
    />
  </div>
</Input>
`;

exports[`uikit/Input should renders correctly with prependValue 1`] = `
<Input
  intent="default"
  name="someInput"
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  prependValue="steam://"
  size="l"
  type="text"
  value="some text"
>
  <div
    className="wrapper size-l"
  >
    <input
      className="input default"
      name="someInput"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="steam://"
      type="text"
      value="steam://some text"
    />
  </div>
</Input>
`;
